### ISSUE TITLE
Car Software Update is showing Installing when Scheduled

### ISSUE THREAD

**Issue by alexbeatle** [alandtse/tesla#356]
![image](https://user-images.githubusercontent.com/56932253/202814076-68597346-5f87-4765-8f1d-9e107ac13254.png)

The Car Software update entity is showing as "installing" when it's actually been scheduled to be installed later in the day.

**Comment by alandtse** [comment]
We'll need the `teslajsonpy` output for the update info to correct this. Can you please enable debugging on it and see if we can capture the value when scheduled?

**Comment by carleeno** [comment]
In `update.py` `in_progress()` it specifically returns `True` when `update_status == "scheduled"`, it looks like it was intentional?

Last night/this morning I have an update available, so I've been collecting response data and tweaking the update component to show the different states (Waiting on Wi-Fi, Downloading, Available, Scheduled, etc) and only enable the Install button when it's able to be installed (Available and Scheduled).

Since it's possible to install immediately while scheduled for a later time, I think it's better to not show it as in-progress while scheduled so that the install button is still available. I'm thinking it could look something like this?
![image](https://user-images.githubusercontent.com/54586926/203565029-6e1e5f23-611e-410d-81a8-cdc8d6bb66f0.png)

**Comment by KirkKirk** [comment]
This is a great option. I'm getting notifications even before the Tesla app.
How can I get info when the software is downloaded and ready to install?
I need it to create automation to trigger the installation in specified conditions (i.e. if received at night and there is enough time before going to work).

**Comment by carleeno** [comment]
> This is a great option. I'm getting notifications even before the Tesla app. How can I get info when the software is downloaded and ready to install? I need it to create automation to trigger the installation in specified conditions (i.e. if received at night and there is enough time before going to work).

This example should work because it will try to install whenever the status string changes in the latest version attribute...but it will only succeed to start the installation when it's ready to install since the install feature is only enabled when it's finished downloading or scheduled. Obviously you'd want to probably add more conditions, such as at home, parked, enough charge, etc.

```yaml
description: "Auto Update"
mode: single
trigger:
  - platform: state
    entity_id:
      - update.mystique_software_update
    attribute: latest_version
condition:
  - condition: time
    before: "05:00:00"
    weekday:
      - mon
      - tue
      - wed
      - thu
      - fri
action:
  - service: update.install
    data: {}
    target:
      entity_id: update.mystique_software_update
```

If you don't want it trying whenever that attribute changes and getting errors that it failed to install when it's still downloading in your logs, then you could get more precise by creating a template sensor that is true when the latest version attribute contains "Available"

**Comment by carleeno** [comment]
@alandtse sorry to ping you, this original issue was fixed by #370 but it's still open.

**Comment by alandtse** [comment]
> > This is a great option. I'm getting notifications even before the Tesla app. How can I get info when the software is downloaded and ready to install? I need it to create automation to trigger the installation in specified conditions (i.e. if received at night and there is enough time before going to work).
> 
> This example should work because it will try to install whenever the status string changes in the latest version attribute...but it will only succeed to start the installation when it's ready to install since the install feature is only enabled when it's finished downloading or scheduled. Obviously you'd want to probably add more conditions, such as at home, parked, enough charge, etc.
> 
> ```yaml
> description: Auto Update
> mode: single
> trigger:
>   - platform: state
>     entity_id:
>       - update.mystique_software_update
>     attribute: latest_version
> condition:
>   - condition: time
>     before: "05:00:00"
>     weekday:
>       - mon
>       - tue
>       - wed
>       - thu
>       - fri
>     after: "00:00:00"
>   - condition: template
>     value_template: >-
>       {{'Available' in state_attr('update.mystique_software_update',
>       'latest_version')}}
> action:
>   - service: update.install
>     data: {}
>     target:
>       entity_id: update.mystique_software_update
> ```
> (updated to include a template condition to only run when status is "Available to install"

This may be good to add to the wiki so people can find it.

**Comment by KirkKirk** [comment]
Thanks, you guys are awesome. It is exactly what I needed and just had an update and it was a successful auto-update!!!!

Cheers
